### PR TITLE
refactor: use superschema for catalog resource/datasource

### PR DESCRIPTION
**Pull request by azrod**
### Description of your changes

fixe #206 

If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### How has this code been tested
```
go test -v -count=1  -run TestAccCatalogDataSource  ./internal/tests/catalog
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests/catalog     16.865s

go test -v -count=1  -run TestAccCatalogResource  ./internal/tests/catalog
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests/catalog     21.888s
```

### CODE REVIEW

File: internal/provider/catalog/catalog_schema.go [hunk 1]
```diff
@@ -48,131 +48,179 @@ type catalogResourceModel struct {
...
+				},
+			},
+			"description": superschema.StringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "The description of the catalog.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Required: true,
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.UseStateForUnknown(),
```
Comment (gaetanars) [inline internal/provider/catalog/catalog_schema.go:124]: Why UseStateForUnknown on a required attribute?

File: internal/provider/catalog/catalog_schema.go [hunk 2]
```diff
@@ -48,131 +48,179 @@ type catalogResourceModel struct {
...
-			PlanModifiers: []planmodifier.String{
-				stringplanmodifier.UseStateForUnknown(),
+			"name": superschema.StringAttribute{
+				Common: &schemaR.StringAttribute{
+					MarkdownDescription: "The name of the catalog.",
+				},
+				Resource: &schemaR.StringAttribute{
+					Required: true,
+					PlanModifiers: []planmodifier.String{
+						stringplanmodifier.UseStateForUnknown(),
```
Comment (gaetanars) [inline internal/provider/catalog/catalog_schema.go:94]: Why UseStateForUnknown on a required attribute?

File: internal/provider/catalog/catalog_schema.go [hunk 3]
```diff
@@ -48,131 +48,179 @@ type catalogResourceModel struct {
...
+					},
+				},
+				DataSource: &schemaD.StringAttribute{
+					Computed: true,
+				},
+			},
+			"href": superschema.StringAttribute{
```
Comment (gaetanars) [inline internal/provider/catalog/catalog_schema.go:131]: Is it a necessary attribute "HREF"?

File: internal/provider/catalog/catalog_schema.go [hunk 4]
```diff
@@ -48,131 +48,179 @@ type catalogResourceModel struct {
...
+						stringplanmodifier.UseStateForUnknown(),
+					},
+				},
+			},
+			"delete_force": superschema.BoolAttribute{
+				Resource: &schemaR.BoolAttribute{
+					Required:            true,
+					MarkdownDescription: "When destroying use `delete_force=True` with `delete_recursive=True` to remove a catalog and any objects it contains, regardless of their state.",
+					PlanModifiers: []planmodifier.Bool{
+						boolplanmodifier.UseStateForUnknown(),
```
Comment (gaetanars) [inline internal/provider/catalog/catalog_schema.go:211]: Why UseStateForUnknown on a required attribute?

File: internal/provider/catalog/catalog_schema.go [hunk 5]
```diff
@@ -48,131 +48,179 @@ type catalogResourceModel struct {
...
+					},
+				},
+			},
+			"delete_recursive": superschema.BoolAttribute{
+				Resource: &schemaR.BoolAttribute{
+					Required:            true,
+					MarkdownDescription: "When destroying use `delete_recursive=True` to remove a catalog and any objects it contains that are in a state that normally allows removal.",
+					PlanModifiers: []planmodifier.Bool{
+						boolplanmodifier.UseStateForUnknown(),
```
Comment (gaetanars) [inline internal/provider/catalog/catalog_schema.go:220]: Why UseStateForUnknown on a required attribute?

File: internal/provider/catalog/catalog_schema.go (hunk 3 above)
Comment (azrod) [reply internal/provider/catalog/catalog_schema.go]: removed

**Review by gaetanars — APPROVED** [review]

**Pull request merged**